server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -117,6 +117,7 @@ void delete(DbSession session, String key, boolean force) {
   }
 
   private void doDelete(DbSession session, QualityProfileDto profile) {
+    db.qualityProfileDao().deleteAllProjectProfileAssociation(profile.getKey(), session);
```

Comment:
Oh yes, now profile to projects associations are in a separate table.
So 2nd question, where are tests testing this ?